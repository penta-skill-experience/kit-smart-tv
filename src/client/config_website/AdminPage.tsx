import * as React from "react";
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import {Grid} from "@mui/material";
import InputLabel from "@mui/material/InputLabel";
import Input from "@mui/material/Input";
import InputAdornment from "@mui/material/InputAdornment";
import IconButton from "@mui/material/IconButton";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import Visibility from "@mui/icons-material/Visibility";
import FormControl from "@mui/material/FormControl";
import Snackbar from "@mui/material/Snackbar";

export const AdminPage = ({oldPassword, newPassword, handleOldPassword, handleNewPassword, handlePasswordChange, children} ) => {
    const [showOldPassword, setShowOldPassword] = React.useState(false);
    const [showNewPassword, setShowNewPassword] = React.useState(false);

    const handleShowOldPassword  = () => {
        setShowOldPassword(!showOldPassword)
    }

    const handleShowNewPassword  = () => {
        setShowNewPassword(!showNewPassword)
    }

    const [successfulBar, setSuccessfulBar] = React.useState(false);
    const [errorBar, setErrorBar] = React.useState(false);

    const handleClose = (event: React.SyntheticEvent | Event, reason?: string) => {
        if (reason === 'clickaway') {
            return;
        }
        setSuccessfulBar(false);
        setErrorBar(false);
    }

    return (
        <Box
        sx={{
            width: 300,
            height: 300,
            }
        }
        >
            <div>
                <Grid container spacing={2} direction="row" justifyContent="center" alignItems="center">
                    <Grid item xs={12}>
                        <FormControl sx={{ m: 1, width: '25ch' }} variant="standard">
                            <InputLabel>Old Password</InputLabel>
                            <Input
                                type={showOldPassword ? 'text' : 'password'}
                                value={oldPassword}
                                onChange={handleOldPassword}
                                endAdornment={
                                    <InputAdornment position="end">
                                        <IconButton
                                            onClick={handleShowOldPassword}
                                        >
                                            {showOldPassword ? <VisibilityOff /> : <Visibility />}
                                        </IconButton>
                                    </InputAdornment>
                                }
                            />
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <FormControl sx={{ m: 1, width: '25ch' }} variant="standard">
                            <InputLabel>New Password</InputLabel>
                                <Input
                                type={showNewPassword ? 'text' : 'password'}
                                value={newPassword}
                                onChange={handleNewPassword}
                                endAdornment={
                                    <InputAdornment position="end">
                                        <IconButton
                                        onClick={handleShowNewPassword}
                                        >
                                        {showNewPassword ? <VisibilityOff /> : <Visibility />}
                                        </IconButton>
                                    </InputAdornment>
                                }
                                />
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <Button
                            onClick={() => {
                                if (handlePasswordChange()) {
                                    setSuccessfulBar(true);
                                } else{
                                    setErrorBar(true);
                                }
                            }}
                            variant="outlined"
                        >
                            Save Password
                        </Button>
                        <Snackbar
                            open={successfulBar}
                            autoHideDuration={6000}
                            onClose={handleClose}
                            message={'Password Saved'}
                        />
                        <Snackbar
                            open={errorBar}
                            autoHideDuration={6000}
                            onClose={handleClose}
                            message={'New Password must at least be 1 character long'}
                        />
                    </Grid>
                    <Grid item xs={12}>
                    </Grid>
                </Grid>
            </div>
        </Box>
    );
}