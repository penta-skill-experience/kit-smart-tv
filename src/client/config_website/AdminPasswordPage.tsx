import * as React from "react";
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';

export const AdminPasswordPage = ({oldPassword, newPassword, handleOldPassword, handleNewPassword, handlePasswordChange, children} ) => {

    return (
        <Box
        sx={{
            width: 300,
            height: 300,
            }
        }
        >
            <div>
                <TextField
                    value={oldPassword}
                    label="Old Password"
                    onChange={handleOldPassword}
                />
                <TextField
                    value={newPassword}
                    label="New Password"
                    onChange={handleNewPassword}
                />
                <Button
                    onClick={handlePasswordChange}
                    variant="outlined"
                >
                    Save Password</Button>
            </div>
        </Box>
    );

}