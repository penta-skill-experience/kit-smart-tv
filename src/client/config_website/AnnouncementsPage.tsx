import * as React from "react";
import {Grid} from "@mui/material";
import TextField from '@mui/material/TextField';
import Button from "@mui/material/Button";
import DeleteIcon from "@mui/icons-material/Delete";

export const AnnouncementsPage = () => {
    const initialMailList = [];
    const [mailList, setMailList] = React.useState(initialMailList);
    const [verUser, setVerUser] = React.useState({
        mail:'',
        name:'',
    });


    const handleMailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, mail:event.target.value});
    };

    const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setVerUser({...verUser, name:event.target.value});
    };

    const handleAddMail = () => {
        if (verUser.name !== '' && verUser.mail !== '') {
            const newUser = {
                mail:verUser.mail,
                name:verUser.name,
            }
            setMailList(mailList.concat(newUser));
        }
        alert('Username and email have to be filled out')
    };

    const handleDeleteUser = (mail) => {
        setMailList(mailList.filter(item => item.mail !== mail));
    }


    return(
        <div>
            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
                <Grid item xs={12}>
                    <h1>Add e-mail as verified: </h1>
                </Grid>
                <Grid item>
                    <TextField
                        label="E-Mail"
                        value={verUser.mail}
                        onChange={handleMailChange}
                    />
                </Grid>
                <Grid item>
                    <TextField
                        label="Name"
                        value={verUser.name}
                        onChange={handleNameChange}
                    />
                </Grid>
                <Grid item>
                    <Button variant="outlined" onClick={handleAddMail}>
                        Add
                    </Button>
                </Grid>
                <Grid item xs={12}>
                    <h1>Currently verified accounts:</h1>
                </Grid>
                <Grid item xs={4}>
                    Email
                </Grid>
                <Grid item xs={2}>
                    Name
                </Grid>
                <Grid item xs={6}></Grid>
                <Grid item xs={12}>
                    <ul>
                        {mailList.map(item => (
                            <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
                                <Grid item xs={4}>
                                    {item.mail}
                                </Grid>
                                <Grid item xs={1}>
                                    {item.name}
                                </Grid>
                                <Grid item xs={1}>
                                    <Button onClick={() => {handleDeleteUser(item.mail)}}>
                                        <DeleteIcon/>
                                    </Button>
                                </Grid>
                            </Grid>
                        ))}
                    </ul>
                </Grid>
            </Grid>
        </div>
    );
}