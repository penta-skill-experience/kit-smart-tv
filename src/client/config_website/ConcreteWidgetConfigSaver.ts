import {WidgetConfigSaver} from "../widget/WidgetConfigSaver";

export class ConcreteWidgetConfigSaver implements WidgetConfigSaver {

    save(rawConfig: Object): void {

        // todo write to persistence
        console.log("save widget config:");
        console.log(rawConfig);
    }
}