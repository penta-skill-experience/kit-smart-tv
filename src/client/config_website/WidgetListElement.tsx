import Radio from "@mui/material/Radio";
import FormControlLabel from "@mui/material/FormControlLabel";
import * as React from "react";
import {Button, Grid} from "@mui/material";
import RadioGroup from '@mui/material/RadioGroup';
import FormControl from '@mui/material/FormControl';
import DeleteIcon from "@mui/icons-material/Delete";
import DialogTitle from '@mui/material/DialogTitle';
import Dialog from '@mui/material/Dialog';
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import DialogActions from "@mui/material/DialogActions";
import {TramScheduleConfigPage} from "./widget_config_pages/TramScheduleConfigPage";
import {RSSFeedConfigPage} from "./widget_config_pages/RSSFeedConfigPage";
import Checkbox from '@mui/material/Checkbox';


export const WidgetListElement = ({item, handlePosition, handleDeleteWidget, handleColorSolid, children}) => {

    return (
        <Grid container spacing={2} direction="row" justifyContent="flex-start" alignItems="center">
            <Grid item xs={1}>
                {item.name}
            </Grid>
            <Grid item>
                <FormControl>
                    <RadioGroup row>
                        {renderPosition({item:item, position:"1", handlePosition: handlePosition})}
                        {renderPosition({item:item, position:"2", handlePosition: handlePosition})}
                        {renderPosition({item:item, position:"3", handlePosition: handlePosition})}
                        {renderPosition({item:item, position:"4", handlePosition: handlePosition})}
                        {renderPosition({item:item, position:"5", handlePosition: handlePosition})}
                    </RadioGroup>
                </FormControl>
            </Grid>
            <Grid item>
                <FormControlLabel control={
                    <Checkbox
                        checked={item.colorSolid}
                        onChange={() => handleColorSolid(item.id, item.colorSolid)}
                    />} label="Solid Color" />
            </Grid>
            <Grid item >
                <DeleteDialogComponent id={item.id} handleDeleteWidget={handleDeleteWidget}/>
            </Grid>
            <Grid item>
                {renderConfigButton({name: item.name})}
            </Grid>
        </Grid>

    );
}

function renderPosition({item, position, handlePosition}) {
    return(
        <FormControlLabel
            label={position}
            control={
                <Radio
                    checked={item.position === position}
                    onChange={
                        () => {
                            handlePosition(item.id, position)
                        }
                    }
                />
            }
        />
    );
}

function DeleteDialogComponent({id ,handleDeleteWidget}) {
    const [open, setOpen] = React.useState(false);


    const handleOpen = () => {
        setOpen(true);
    };

    const handleDisagree = () => {
        setOpen(false);
    };

    const handleAgree = () => {
        handleDeleteWidget(id);
        setOpen(false);
    };

    return(
        <Grid item>
            <Button onClick={handleOpen}>
                <DeleteIcon/>
            </Button>
            <Dialog onClose={handleDisagree} open={open}>
                <DialogTitle>Delete this widget?</DialogTitle>
                <DialogContent>
                    <DialogContentText>
                        The changes can't be restored and have to be manually recreated.
                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button variant="outlined" onClick={handleDisagree}>Disagree</Button>
                    <Button variant="outlined" onClick={handleAgree}>
                        Agree
                    </Button>
                </DialogActions>
            </Dialog>
        </Grid>
    );
}

function renderConfigButton({name}) {
    if (name === "Tram Schedule") {
        return (
            <TramScheduleConfigPage/>
        );
    } else if (name === "RSS feed") {
        return (
          <RSSFeedConfigPage/>
        );
    }
}