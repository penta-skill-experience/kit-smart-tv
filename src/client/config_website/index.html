<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Admin Interface</title>
    <meta content="width=device-width, initial-scale=1" name="viewport">
    <link
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            rel="stylesheet"
    />
    <style>
        #root {
            height: 100vh;
            width: 100vw;
        }
    </style>
</head>
<body>
<div id="root">
    <!-- react output will go inside here -->
</div>
</body>
</html>
