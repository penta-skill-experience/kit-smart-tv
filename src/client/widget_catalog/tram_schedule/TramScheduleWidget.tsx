import * as React from "react";
import {Widget} from "../../widget/Widget";
import {TramScheduleDisplayComponent} from "./TramScheduleDisplayComponent";
import {ConfigComponent} from "../../widget/ConfigComponent";
import {TramScheduleConfigComponent} from "./TramScheduleConfigComponent";
import {DisplayComponent} from "../../widget/DisplayComponent";
import * as TramScheduleConfig from "./TramSchedule.json";

export class TramScheduleWidget implements Widget {

    getConfigComponentClass(): typeof ConfigComponent {
        return TramScheduleConfigComponent;
    }

    getDisplayComponentClass(): typeof DisplayComponent {
        return TramScheduleDisplayComponent;
    }

    getTitle(): string {
        return "Tram Schedule";
    }

    isConfigurable(): boolean {
        return true;
    }

    getDefaultRawConfig(): Object {
        return {
            stop: "",
            count: TramScheduleConfig.DEFAULT_ITEM_COUNT,
        };
    }
}