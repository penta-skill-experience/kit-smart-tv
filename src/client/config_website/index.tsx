import "./index.css";
import * as ReactDOM from "react-dom";
import * as React from "react";
import {WidgetLoader} from "../widget/WidgetLoader";
import {ConcreteWidgetConfigSaver} from "./ConcreteWidgetConfigSaver";

ReactDOM.render(
    // this is just an example to demonstrate how widgets are instantiated
    // todo: remove this.
    <div>
        <h1>Hello Admin!</h1>
        <div>
            {new WidgetLoader().getWidget("rss-feed").createConfigComponent({url: "www.example.com"}, new ConcreteWidgetConfigSaver())}
        </div>
    </div>,
    document.getElementById("root")
);