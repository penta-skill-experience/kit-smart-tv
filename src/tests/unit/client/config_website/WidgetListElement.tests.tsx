import {shallow, configure} from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import * as React from "react";
import toJson from "enzyme-to-json";
import {WidgetListElement, DeleteDialogComponent} from "../../../../client/config_website/WidgetListElement";
import {TramScheduleWidget} from "../../../../client/widget_catalog/tram_schedule/TramScheduleWidget";

configure({adapter: new Adapter()});

let wrapper;
const method = () => {};
const testWidget = new TramScheduleWidget();
const testWidgetData = {
    widgetId: "time",
    location: 2,
    rawConfig: undefined,
};

const testItem = {
    id:0,
    position:'2',
    widgetNameText:'name',
    widget: testWidget,
    widgetData:testWidgetData,
}

const handlePositionMock = jest.fn();
const handleDeleteWidgetMock = jest.fn();

describe("Widget list element tests", () => {

    beforeEach(() => {
            wrapper = shallow(
                <WidgetListElement
                    item={testItem}
                    handlePosition={handlePositionMock}
                    handleDeleteWidget={handleDeleteWidgetMock}
                    handleRawConfigSave={method}
                >
                </WidgetListElement>
            );
    });

    test("snapshot test for widget list element", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    test("test radio on change function", () => {
        const newWrapper = shallow(wrapper.find("ForwardRef(FormControlLabel)").at(0).prop("control"));
        newWrapper.find("MuiRadioRoot").at(0).simulate("change");
        expect(handlePositionMock).toHaveBeenCalled();
    })

    test("handleDeleteWidget test", () => {
        wrapper.find("DeleteDialogComponent").prop("handleDeleteWidget")({target:{}});
        expect(handleDeleteWidgetMock).toHaveBeenCalled();
    });

});

describe("Delete Dialog Component tests", () => {
    let deleteDialogComponentWrapper;
    beforeEach(() => {
        deleteDialogComponentWrapper = shallow(
            <DeleteDialogComponent
                id={""}
                handleDeleteWidget={method}
            />
        );
    });

    test("snapshot test for DeleteDialogComponent", () => {
        expect(toJson(deleteDialogComponentWrapper)).toMatchSnapshot();
    });

    test("handleOpen test", () => {
        deleteDialogComponentWrapper.find("ForwardRef(Button)").at(0).simulate("click");
        expect(deleteDialogComponentWrapper.find("ForwardRef(Dialog)").prop("open")).toEqual(true);
    });

    test("handleDisagree test", () => {
        deleteDialogComponentWrapper.find("ForwardRef(Button)").at(1).simulate("click");
        expect(deleteDialogComponentWrapper.find("ForwardRef(Dialog)").prop("open")).toEqual(false);
    });

    test("handleAgree test", () => {
        deleteDialogComponentWrapper.find("ForwardRef(Button)").at(2).simulate("click");
        expect(deleteDialogComponentWrapper.find("ForwardRef(Dialog)").prop("open")).toEqual(false);
    });
})