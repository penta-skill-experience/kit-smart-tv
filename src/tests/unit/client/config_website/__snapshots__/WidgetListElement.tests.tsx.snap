// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Dialog Component tests snapshot test for DeleteDialogComponent 1`] = `
<ForwardRef(Grid)
  item={true}
>
  <ForwardRef(Button)
    onClick={[Function]}
  >
    <Memo(ForwardRef(DeleteIcon)) />
  </ForwardRef(Button)>
  <ForwardRef(Dialog)
    onClose={[Function]}
    open={false}
  >
    <ForwardRef(DialogTitle)>
      Delete this widget?
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogContent)>
      <ForwardRef(DialogContentText)>
        The changes can't be restored and have to be manually recreated.
      </ForwardRef(DialogContentText)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button)
        onClick={[Function]}
        variant="outlined"
      >
        Disagree
      </ForwardRef(Button)>
      <ForwardRef(Button)
        onClick={[Function]}
        variant="outlined"
      >
        Agree
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</ForwardRef(Grid)>
`;

exports[`Widget list element tests snapshot test for widget list element 1`] = `
<ForwardRef(Grid)
  alignItems="center"
  container={true}
  direction="row"
  justifyContent="flex-start"
  spacing={2}
>
  <ForwardRef(Grid)
    item={true}
    xs={1}
  >
    Tram Schedule
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
  >
    <ForwardRef(FormControl)>
      <ForwardRef(RadioGroup)
        row={true}
      >
        <ForwardRef(FormControlLabel)
          control={
            <ForwardRef(Radio)
              checked={false}
              onChange={[Function]}
            />
          }
          label={1}
        />
        <ForwardRef(FormControlLabel)
          control={
            <ForwardRef(Radio)
              checked={true}
              onChange={[Function]}
            />
          }
          label={2}
        />
        <ForwardRef(FormControlLabel)
          control={
            <ForwardRef(Radio)
              checked={false}
              onChange={[Function]}
            />
          }
          label={3}
        />
        <ForwardRef(FormControlLabel)
          control={
            <ForwardRef(Radio)
              checked={false}
              onChange={[Function]}
            />
          }
          label={4}
        />
        <ForwardRef(FormControlLabel)
          control={
            <ForwardRef(Radio)
              checked={false}
              onChange={[Function]}
            />
          }
          label={5}
        />
      </ForwardRef(RadioGroup)>
    </ForwardRef(FormControl)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
  >
    <DeleteDialogComponent
      handleDeleteWidget={[MockFunction]}
      id={0}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
  >
    <WidgetConfigPage
      configComponentClass={[Function]}
      save={[Function]}
    />
  </ForwardRef(Grid)>
</ForwardRef(Grid)>
`;
